src/atlas/interpolation/method/knn/KNearestNeighbours.cc

```diff
@@ -62,11 +62,19 @@ void KNearestNeighbours::do_setup(const Grid& source, const Grid& target, const
     do_setup(functionspace(source), functionspace(target));
 }
 
+void KNearestNeighbours::do_setup(const FunctionSpace& source, const FunctionSpace& target, const Cache& cache) {
+    if (interpolation::MatrixCache(cache)) {
+        setMatrix(cache);
+        source_ = source;
+        target_ = target;
+        buildPointSearchTree(source);
```

Comment:
Building the PointSearchTree should not be needed when the cache contains a matrix.

On the other hand when there's no matrix in the cache this sets up nothing. That needs fixing.
```c++
if(interpolation::MatrixCache(cache)) { ... } 
else { 
  THIS SITUATION
}
```